tests/perf/test_accuracy.py

```diff
@@ -2,142 +2,255 @@
             model=model,
             task=Task.CLASSIFICATION,
             preprocess_fn=self.preprocess_fn,
-            explain_mode=ExplainMode.WHITEBOX,
+            postprocess_fn=self.postprocess_fn,
+            explain_mode=explain_mode,
+            explain_method=explain_method,
+            embed_scaling=True,
         )
+        kwargs = {}
+        if explain_method in BlackBoxXAIMethods:
+            # TODO: Make Preset configurable as well
+            kwargs.update({"preset": Preset.SPEED})
+        return kwargs
+
+    @pytest.fixture(autouse=True)
+    def setup(self, fxt_output_root, fxt_data_root, fxt_dataset_parameters):
+        self.data_dir = fxt_data_root
+        # self.output_dir = fxt_output_root
+        self.output_dir = Path("/home/gzalessk/code/openvino_xai/tests/perf/validation_results/5000_subset_42_seed")
+        self.supported_num_classes = {1000: 1000}
+
+        self.setup_dataset(fxt_dataset_parameters)
+        self.dataset_name = "coco" if self.dataset_type == DatasetType.COCO else "voc"
+
+    @pytest.mark.parametrize("model_id", TEST_MODELS)
+    @pytest.mark.parametrize("explain_method", EXPLAIN_METHODS)
+    def test_explainer_images(self, model_id, explain_method):
+        self.model_name = model_id
+        self.data_metric_path = self.output_dir / self.model_name / explain_method.value
+        os.makedirs(self.data_metric_path, exist_ok=True)
+
+        model, model_cfg = self.setup_model(self.data_dir, self.model_name)
+        self.setup_process_fn(model_cfg)
+        black_box_kwargs = self.setup_explainer(model, explain_method)
 
         self.pointing_game = PointingGame()
         self.auc = InsertionDeletionAUC(model, self.preprocess_fn, self.postprocess_fn)
-        self.adcc = ADCC(model, self.preprocess_fn, self.postprocess_fn, self.explainer)
+        self.adcc = ADCC(model, self.preprocess_fn, self.postprocess_fn, self.explainer, **black_box_kwargs)
 
-    def setup_dataset(self, data_root: str, ann_path: str):
-        self.dataset_type = define_dataset_type(data_root, ann_path)
-        self.channel_format = "RGB" if self.dataset_type in [DatasetType.VOC, DatasetType.COCO] else "None"
+        records = []
+        explained_images = 0
+        experiment_start_time = time()
+        max_num = len(self.dataset)
+        batch_size = 5000
 
-        if self.dataset_type == DatasetType.COCO:
-            self.dataset = datasets.CocoDetection(root=data_root, annFile=ann_path)
-            self.dataset_labels_dict = {cats["id"]: cats["name"] for cats in self.dataset.coco.cats.values()}
-            self.anns_to_gt_bboxes = coco_anns_to_gt_bboxes
-        elif self.dataset_type == DatasetType.VOC:
-            self.dataset = datasets.VOCDetection(root=data_root, download=False, year="2012", image_set="val")
-            self.dataset_labels_dict = None
-            self.anns_to_gt_bboxes = voc_anns_to_gt_bboxes
+        for lrange in range(0, max_num, batch_size):
```

Comment:
This is a good place to have a progress bar

Reply:
Fixed in [c3ed6c9](https://github.com/openvinotoolkit/openvino_xai/pull/73/commits/c3ed6c9d8a358ca9196d5ef94c6e9cffe3f2d994)